spec/controllers/hub/messages_controller_spec.rb

```diff
@@ -28,6 +28,60 @@
         end
       end
 
+      context "viewing communication preferences" do
+        render_views
+        context "without way to communicate" do
+          before do
+            intake.update(sms_notification_opt_in: "no", email_notification_opt_in: "no", email_address: "", sms_phone_number: nil)
```

Comment:
why do we save blank email addresses as empty strings and blank sms phone numbers as nil?

Reply:
It's a validation problem -- the create client form is allowing people to send in empty emails but click that they are opted into email. That's funky, but i wanted to replicate that funky behavior in the tests to make sure this catches that case.